### PR TITLE
[Submission] ChinYun, KinWai

### PR DESCRIPTION
**Pull request by yoyololicon**
**ChinYun-KinWai**

Dear mdx workshop committee,

I request a review for the following MDX submission:

### **Title: `Danna-Sep: Unite to Separate Them All`**  

**Author(s): Chin-Yun Yu, Kin-Wai Cheuk**  

https://www.aicrowd.com/challenges/music-demixing-challenge-ismir-2021/teams/Kazane_Ryo_no_Danna

**Category**

* [x] `POSTERS` _(+Abstract: 250 Words+one page pdf)_
  > As a participant of the MDX challenge, please know that your poster would be automatically accepted after a minimal prescreening. Hence, don't forget to provide us with the link to your team in the AIcrowd website.
  > You are also welcome to submit us some unrelated abstract, in case your interest would be in presenting some method or idea you would just like to discuss with the participants passing by.  
  > => You *don't* need to include your poster in the PR. Just be ready to have it for D-day and show it to the audience through screen sharing.
* [ ] `LONG TALK`
  > A long talk will last about 20-30min, including questions. This can be about some recent paper you published, or may also present some original research and investigations you did. If you pick this category, please provide us with an extended abstract of the talk, or directly with the slides if you have them available, even in an unfinished state, at your convenience. 
* [ ] `DISCUSSIONS`
  > This category is intended to stimulate fruitful discussions and collaborations. If you pick it, it means you are willing to chair some open discussion with the participants, mainly around one particular aspect of music demixing research. For that purpose, you should provide a minimal abstract as well as a list of a few questions you would like the participants to discuss about. Don't hesitate to propose ideas, We will possibly contact you in case we identify several similar topics that were proposed!

**Preferred Timezone**

* [x] A 11:00 PM (PDT) to 9:00 AM PDT
* [ ] B 11:00 AM (PDT) to 9:00 PM PDT

  > The preferred timezone will help us to organize the day schedule. You should select at least one, but can perfectly select both if you don't care. We will organize the programme so that to maximally fit your preferences.
---

**ORGANISATION COMMITTEE (do not fill out)**

* [X] Editor acknowledgment
* [X] Reviewer 1 
* [x] Reviewer 2
* [X] Review 1 decision [accept]
* [ ] Review 2 decision [accept/reject]
* [ ] Editor decision [accept/reject]

### CODE REVIEW

File: paper.md [hunk 1]
```diff
@@ -1,139 +1,102 @@
...
+We trained the model using the same time-domain loss as the original X-UMX, but modified the frequency-domain loss for $J$ sources as follows:
 
-The expected length for submissions in this category is around one page, excluding references.
-It would be nice to have some illustration if applicable.
+$$\mathcal{L}_{MSE}^J = \sum_{j=1}^J\sum_{t,f}|Y_j(t, f) - \hat{Y}_j(t, f)|^2$$
 
+where $Y_j(t, f)$ and $\hat{Y}_j(t, f)$ are ground-truth and estimated time-frequency representations for the $j$-th source, respectively. 
+That is, instead of taking norm of the absolute value as in the original X-UMX, we calculated Euclidean norm in the complex domain.
+Also, we incorporated Multichannel Wiener Filtering (MWF)[@antoine_liutkus_2019_3269749] into our training pipeline in order to train our model in an end-to-end fashion. We initialized our modified X-UMX with the official pre-trained X-UMX weights^[<https://zenodo.org/record/4740378/files/pretrained_xumx_musdb18HQ.pth>] and continued training for approximately 70 epochs with a batch size of four.
```
Comment (aliutkus) [inline paper.md:51]: Out of curiosity: how many iterations did you use for this ? Did you use softmask ? (what were the parameters)

File: paper.md [hunk 2]
```diff
@@ -1,139 +1,102 @@
...
+| X-UMX   | 0.2   | 0.1  | 0     | 0.2    | TF | 136
+| U-Net   | 0.2   | 0.17 | 0.5   | 0.4    | TF | 61
+| Demucs  | 0.6   | 0.73 | 0.5   | 0.4    | T | 733
 
-## Mathematics
+All models were trained on the training set of musdb18-hq [@musdb18-hq] using an Adam optmizier[@kingma2014adam]. 
 
-Single dollars ($) are required for inline mathematics e.g. $f(x) = e^{\pi/x}$
+# Separation performances
+For a fair comparison, we trained all the models with musdb18-hq [@musdb18-hq] and performed the evaluation using the compressed version of the dataset [@musdb18].
```
Comment (aliutkus) [inline paper.md:78]: you trained with musdb-hq-train and evaluated with musdb18-test, right ?

File: paper.md [hunk 3]
```diff
@@ -1,139 +1,102 @@
...
 
-$$\Theta(x) = \left\{\begin{array}{l}
-0\textrm{ if } x < 0\cr
-1\textrm{ else}
-\end{array}\right.$$
+As can be seen from the table, our modified X-UMX gained an extra 0.27 dB on average SDR compared to the original X-UMX.
+The enhanced Demucs outperformed the original model by 0.03 dB of SDR, despite the fact that the shift trick was not applied.
+Notably, Danna-Sep surpassed both the original and enhanced Demucs by a large margin (+0.53 dB on average SDR).
+Altogether, the results demonstrate the efficacy of the proposed fusion method in addition to our modifications to the training scheme and architecture.
+The proposed framework, however, is more reliant on computing power due to the nature of model fusion, which we would like to address in furture work.
```
Comment (aliutkus) [inline paper.md:95]: this seems quite intrinsic to the method. Do you have any idea how ?

**Review by aliutkus — COMMENTED** [review]
natural and greatly lead fusion of models. interesting !

File: paper.md (hunk 1 above)
Comment (yoyololicon) [reply paper.md:51]: Same settings as the original X-UMX when doing inference, one iteration without soft mask.

File: paper.md (hunk 2 above)
Comment (yoyololicon) [reply paper.md:78]: Yes. The reason that we evaluated on musdb18 is to help readers compare the scores with previous works.

File: paper.md (hunk 3 above)
Comment (yoyololicon) [reply paper.md:95]: We're planning to port the model into onnx format, which should hasten the inference process a little further than using TorchScript . Other possible directions would be model compression or quantization, but we haven't looked deep into it yet.

**Comment by pseeth** [comment]
Sure I can review it.

**Review by pseeth — COMMENTED** [review]
Nice idea with solid results! The boosts in SDR are encouraging but I want to point out a few ways that I think this could be improved, experimentally, especially for a bigger submission down the road.

If the idea is the prove that multi-representation approaches are better than single-representation approaches, then I'm not sure if the paper argues it in an apples-to-apples way. There are two spectrogram approaches, for one waveform approach, and they all have very different architectures. One could argue that the improvement comes from combining convolutional and recurrent architectures, rather than spectrogram and waveform inputs. Or it could come from a higher receptive field in the spectrogram architectures (because of the higher hop size), versus the lower receptive field of the waveform architectures.

One way to get at this would be to try to keep more things fixed - like keeping the architecture the same and varying the input representation, and building up to the final configuration with more ablations.

On the other hand, the ensemble approach looks solid and provides clear improvement, at the cost of increased system complexity. I like that, in this approach, the models are combined simply using just a linear approach. However, I wonder if you could improve this even further by using a non-linear combination of models (as an aside, my co-authors and I did something like this with non-deep-learning algorithms [here](https://www.researchgate.net/profile/Fatemeh-Pishdadian/publication/321788814_Predicting_algorithm_efficacy_for_adaptive_multi-cue_source_separation/links/5ad56de5458515c60f5472f1/Predicting-algorithm-efficacy-for-adaptive-multi-cue-source-separation.pdf)). Nowadays, I would just throw a stack of convolutions on the output from each of the 3 models and then train it end-to-end, and call it a day.

Finally, you could consider connections between models before they input their final decisions, so they can be ensembled more effectively.

Overall, I like the underlying idea! The above are just things that I think could be improved for a bigger submission down the line. Nice work! Decision: accept.

Thanks for sharing!

**Comment by yoyololicon** [comment]
@pseeth 
Thanks for the comments and suggestions! 
I agree that our experiments can be done in a more systematic way (partly due to the limited time when participating the challenge), and we'll definitely look into this direction in the future.

**Comment by aliutkus** [comment]
To all participants:
please note that you should update your submission by November 6th 23:00 UTC

Thank you very much

**Comment by yoyololicon** [comment]
Hi @aliutkus,
I see the checkbox of reviewer 2 was checked, and I'm not sure whether our poster is accepted. Do we have a second reviewer?

**Comment by faroit** [comment]
@yoyololicon  Great to have you presenting a poster. I couldn't find you among the workshop registrants at ISMIR. Please register within the next day so that I can add you to the schedule.

**Comment by aliutkus** [comment]
Yes, accepted ! Sorry for the delay

**Comment by faroit** [comment]
👏 The submission has been published on midspace (the ISMIR conference platform). The submissions will also be hosted on the workshop website tomorrow. 

The schedule can be seen there or [here](https://docs.google.com/spreadsheets/d/1PaksFcHWUfNPMg-XloA3CJtrvb9uMU0erCFVyvFi9B4/edit?usp=sharing). Please let us know if there is anything missing. Otherwise we see each other on Friday/Saturday!

**Pull request closed without merging**